### ISSUE TITLE
Laravel 8 Factories

### ISSUE THREAD

**Issue by carlobeltrame** [NoelDeMartin/laravel-cypress#5]
Laravel 8 has arrived, and rewritten factories from scratch: https://laravel.com/docs/8.x/releases#laravel-8

Do you have any idea how to use the new class-based factories with this package?

**Comment by NoelDeMartin** [comment]
Hi there, thanks for opening an issue!

I haven't looked into this in detail, but looking at the docs and from what I saw at Laracon it seems like it's a very different approach. The good part is that the old syntax is still supported installing the [laravel/legacy-factories](https://github.com/laravel/legacy-factories) package.

I recently added new functionality to define [custom Cypress commands](https://github.com/NoelDeMartin/cypress-laravel#define-your-own-commands) that communicate with your Laravel app. That should be useful if you want to create your own factory commands compatible with Laravel 8, and you can make them as flexible or as rigid as you like.

The current implementation only supports attributes and count, so it would be easy to keep the same syntax. But I could use this opportunity to add support for more. I don't think I'll do it for a while because I have other priorities. Expect at least a couple of months, maybe even next year. But here's an initial brainstorming on how I would approach this. Of course, anyone is welcome to implement this or a different proposal and open a PR :).

So, the problem with the new syntax is that instead of calling a single method with different arguments, it uses method chaining (which already happened with states before, but they are not supported in this package yet). My initial idea is to support a syntax like this, and use a Proxy under the hood to communicate the whole method call chain (similar to Livewire Javascript Components):

```js
cy.create('User', factory => factory.hasPosts(3).create())
    .then(user => {
        // ...
    });
```

This gets hairy if you want to use more than one model, so maybe this could be supported as well:

```js
cy.create(['User', 'Post'], (userFactory, postFactory) => {
        return userFactory.has(postFactory.count(3)).create();
    })
    .then(user => {
        // ...
    });
```

**Comment by NoelDeMartin** [comment]
Ok, I'm trying something for [hacktoberfest](https://hacktoberfest.digitalocean.com/) this year :).

If anyone participating in hacktoberfest wants to help, this is a relatively simple issue for Laravel developers. All you need to do is update [this line](https://github.com/NoelDeMartin/laravel-cypress/blob/master/src/Http/Controllers/CypressController.php#L49) to support Laravel 8 factories, in a backwards compatible way.

What I would do is check if the model class has the new [factory method](https://laravel.com/docs/8.x/database-testing#creating-models) defined and call it. If it doesn't exist, fall back to using the current implementation.

----

As per my comment above, that's a different story and anyone is welcome to open a PR as well. But I don't expect it given the complexity, so I'll open a different issue for that once this is closed.

**Comment by NoelDeMartin** [comment]
Solved with #6 and released in v0.2.1, I opened a different issue to keep track of the improvements in mentioned.
